apps/definition_kafka/lib/kafka/topic/destination.ex

```diff
@@ -0,0 +1,155 @@
+defmodule Kafka.Topic.Destination do
+  use GenServer, shutdown: 30_000
+  use Properties, otp_app: :definition_kafka
+  require Logger
+
+  getter(:dlq, default: Dlq)
+
+  @spec start_link(Destination.t(), Destination.init_opts()) ::
+          {:ok, Destination.t()} | {:error, term}
+  def start_link(topic, opts) do
+    GenServer.start_link(__MODULE__, {topic, opts})
+    |> Ok.map(&%{topic | pid: &1})
```

Comment:
is the `{:ok, pid}` still returned to the caller here or is it swallowed by merging it into the topic state?

Reply:
`Destination.start_link/2` calls for `{:ok, Destination.t()}` to be returned. So the pid is wrapped in the struct being returned. @bbalser may be changing that soon.

Reply:
if this is going to not be a process soon that makes sense; from what i can tell this should still work from a supervision strategy since it's linked to the caller and that should be under a dynamic supervisor but it just felt like a trap only burying that pid in another process's state.